mantidproject/mantid
# Create and Embed the RegionSelector In the Reflectometry GUI

**Description of work.**
Takes the data_view portion of the SliceViewer, puts it into it's own little widget, and then embeds it into the Reflectometry GUI. Also wires it up with the instrument viewer so that only the selected detectors are plotted on the viewer. 

**To test:**
1. Open the Refl GUI (Interfaces -> Reflectometry -> ISIS Reflectometry)
2. Switch to the Preview tab
3. Load the `INTER45455_inst.nxs` data file (available from the Testing Instructions on #31069)
4. Select the rectangle tool and highlight a portion of the instrument
5. Observe the sliceviewer be created and added to the GUI
6. Move the rectangle around and see the SV update its plot, showing only the highlighted spectra and counts. 

Part of #33735 


*This does not require release notes* because **this is part of an ongoing piece of work.**




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
